### ISSUE TITLE
Monitor package downloads over time

### ISSUE THREAD

**Issue by michaelrambeau** [bestofjs/bestofjs-webui#78]
## Request

This is a request from @Unitech (Alexandre Strzelewicz) the creator of [PM2](https://github.com/Unitech/pm2) package:

> What is missing from my point of view is to be able to monitor downloads over time, I use often this app to track download progress: https://npm-stat.com/charts.html?package=pm2. The download status is public data without rate limitation so might worth the integration?

## What we could do

In _Best of JavaScript_ project details pages, inside the "Package" block (see below), we could add a "Downloads" section that would show a graph of the number of download by month, over the last year.

![image](https://user-images.githubusercontent.com/5546996/68094043-b9cea300-fedf-11e9-865e-91800e7da4cc.png)

## How to grab data

Which API to use to grab the number of downloads?

NPMS API provides only the total number of download over different periods of time: 

- last year
- last 6 months
- last 3 months
- last month
- last week
- last 24 hours 

For example:  https://api.npms.io/v2/package/pm2

However it does not return the number of downloads month by month, so we can't use this data to build a graph.

`npm-stat.com` provides the number of downloads day by day, for example: https://npm-stat.com/api/download-counts?package=pm2&from=2018-11-03&until=2019-11-03

So we could use this data to build both a daily and monthly graph.

But this is not an official API end-point, I think it's better to ask `npm-stats.com` creator if we could use their data.

What do you think @Unitech:

- Would the monthly graph I described be useful for your use-case?
- Can you see an open API we could use to grab data?

Thank you in advance for your feedback!

**Comment by Unitech** [comment]
About accessing stats from npm-stats, you could ping @pvorb 
I like to be able to switch between daily/weekly/monthly/yearly downloads stats, but monthly by default would be enough.

**Comment by pvorb** [comment]
You should probably prefer to use NPM's API directly. See <https://github.com/npm/registry/blob/master/docs/download-counts.md>.

**Comment by michaelrambeau** [comment]
Thank you @pvorb for the advice, this is exactly what we need to show the variations of the number of downloads.

Example of request: https://api.npmjs.org/downloads/range/last-year/redux

```json
{
  "start":"2018-11-08",
  "end":"2019-11-07",
  "package":"redux",
  "downloads":[
    {"downloads":436876,"day":"2018-11-08"},
    
    {"downloads":732112,"day":"2019-11-07"}
  ]
}
```

We will try to integrate this data as soon as possible.

**Comment by michaelrambeau** [comment]
Hello Alexandre @Unitech 
Sorry it took me 5 months to implement the feature (!) because I had other priorities (make the home page more attractive for example) but the chart of the monthly downloads is available!

Could you check this temporary URL: https://develop--bestofjs.netlify.com/projects/pm2

![image](https://user-images.githubusercontent.com/5546996/78471867-5b983b80-776f-11ea-8ae6-d762e2406614.png)

According to this metric, it seems that `pm2` package is doing well!

I will deploy very soon if the feature is OK for you.
Any feedback is welcome (including what you think about the new homepage too).
Thank you in advance!

**Comment by Unitech** [comment]
That looks great! Thank you for this feature

**Comment by michaelrambeau** [comment]
The feature was released on 2020-04-06 (v0.22)
